packages/headless/src/controllers/insight/quickview/headless-insight-quickview.test.ts

```diff
@@ -46,90 +57,44 @@ describe('Insight Quickview', () => {
     });
   });
 
-  it('exposes a subscribe method', () => {
-    expect(quickview.subscribe).toBeTruthy();
-  });
-
-  describe('#fetchResultContent', () => {
-    const uniqueId = '1';
-    const requestedOutputSize = 0;
-
-    beforeEach(() => {
-      options.result = buildMockResult({uniqueId});
-      initQuickview();
-
-      quickview.fetchResultContent();
-    });
-
-    it('dispatches a #updateContentURL action with the result uniqueId', () => {
-      const action = engine.findAsyncAction(updateContentURL.pending);
-
-      expect(action?.meta.arg.uniqueId).toBe(uniqueId);
-      expect(action?.meta.arg.requestedOutputSize).toBe(requestedOutputSize);
-      expect(action?.meta.arg.path).toBe('/quickview');
-    });
-
-    it('dispatches a document quickview click event', () => {
-      const result = buildMockResult();
-      const thunk = logDocumentQuickview(result);
-      const action = engine.findAsyncAction(thunk.pending);
-
-      expect(action).toBeTruthy();
-    });
+  it('calls #buildCoreQuickview and returns its results', () => {
+    expect(mockedBuildCoreQuickview).toHaveBeenCalledWith(
+      engine,
+      {options: {...options, onlyContentURL: true}},
+      expect.any(Function),
+      '/quickview',
+      expect.any(Function)
+    );
+    expect(mockedBuildCoreQuickview).toHaveLastReturnedWith(quickview);
   });
 
-  it(`when configured result uniqueId matches the uniqueId in state,
-  #state.content returns the content in state`, () => {
-    const uniqueId = '1';
-    const content = '<div></div>';
+  it('#buildResultPreviewRequest calls #buildInsightResultPreviewRequest and returns its results', () => {
+    const mockedBuildInsightResultPreviewRequest = jest.mocked(
+      buildInsightResultPreviewRequest
+    );
 
-    engine.state.resultPreview = buildMockResultPreviewState({
-      uniqueId,
-      content,
-    });
-    options.result = buildMockResult({uniqueId});
-    initQuickview();
-
-    expect(quickview.state.content).toEqual(content);
-  });
-
-  it(`when configured result uniqueId matches the uniqueId in state,
-  #state.content returns an empty string`, () => {
-    engine.state.resultPreview = buildMockResultPreviewState({
-      uniqueId: '1',
-      content: '<div></div>',
-    });
-    options.result = buildMockResult({uniqueId: '2'});
-    initQuickview();
-
-    expect(quickview.state.content).toEqual('');
-  });
+    const someHtmlRequestOptions: HtmlRequestOptions = {
+      uniqueId: 'some-id',
+      requestedOutputSize: 0,
+    };
 
-  [true, false].forEach((testValue) => {
-    it(`when the result #hasHtmlVersion is ${testValue} #state.resultHasPreview should be ${testValue}`, () => {
-      options.result = buildMockResult({hasHtmlVersion: testValue});
-      initQuickview();
+    mockedBuildCoreQuickview.mock.calls[0][2](state, someHtmlRequestOptions);
```

Comment:
At a glance, this is pretty cryptic to me and makes it dififcult to quickly grasp how the test actually works.

Looking at the Jest documentation and at the `buildCoreQuickview` function signature, I gather that we're calling the third argument of the first call made to the mocked `buildCoreQuickview`.

How about we do something like:

```suggestion
    const mockedBuildResultPreviewRequest = mockedBuildCoreQuickview.mock.calls[0][2];
    mockedBuildResultPreviewRequest(state, someHtmlRequestOptions);
```

in these kinds of tests?

This comments applies to all similar instances.

Reply:
> Looking at the Jest documentation and at the buildCoreQuickview function signature, I gather that we're calling the third argument of the first call made to the mocked buildCoreQuickview.

Yes

I'll work something out based on your suggestions (I don't want to use mock when it's not a mock, and the 3rd param is not a mock)